### ISSUE TITLE
Stats are not calculated when attacking

### ISSUE THREAD

**Issue by extazz17** [superseva/fallout#71]
Hello,
I created the items and linked them to the right attributes, however when I roll my dice to attack, I have the characteristic (strength) which is taken but I have nothing in the skill (Armes Corps à Corps) and the advantage is not checked.
Is this a bug or a configuration issue?

Thank you

![image](https://github.com/superseva/fallout/assets/55485926/12101bd8-1b20-4490-a90f-ae1b5af82818)
![image](https://github.com/superseva/fallout/assets/55485926/09a31cc5-6519-473b-a30e-3ae41b959a44)
![image](https://github.com/superseva/fallout/assets/55485926/370c62f6-0b29-4436-8456-211e52eb4ef3)

**Comment by superseva** [comment]
the skill name on the sheet is not the same as on the weapon.
On the weapon sheet you have `Armes De Corps A Cor....` and on the character sheet you named it `Armes de CaC`
Those need to be the same names. Did you change translation file ?

**Comment by superseva** [comment]
see here, they have the same name skill and the skill on the weapon.
![chrome_BRCZD8UIyh](https://github.com/superseva/fallout/assets/4093045/f41b31b5-6d9f-4e23-a270-a92216c8a537)

**Comment by extazz17** [comment]
I installed the good translation module (previous was a old one).
Now i have the exact same translation but it doesn't work : 
![image](https://github.com/superseva/fallout/assets/55485926/1456691f-cf6a-4f3d-93e1-8f1dceb224e1)
![image](https://github.com/superseva/fallout/assets/55485926/171ea516-0dc6-4703-b36b-a96533995c98)
![image](https://github.com/superseva/fallout/assets/55485926/6d5a2f1e-689a-42af-b569-505bd05dfd09)

**Comment by extazz17** [comment]
Note : I tested severals things and if i don't use any translation module but the native FR translation from your module : 

- I set a value to the `Melee Weapons`
![image](https://github.com/superseva/fallout/assets/55485926/0e2fdd22-5875-4189-bcba-37fa02143c3a)
- In the weapon card, i still have FR translations (natives ?)
![image](https://github.com/superseva/fallout/assets/55485926/0cd83f1f-5c2e-4204-a9a1-4983acfa695a)
- The link is ok
![image](https://github.com/superseva/fallout/assets/55485926/386e3eb4-1fea-4c18-a3f9-b02b6e956c03)

**BUT**, if i update the default label from "Melee Weapons" to "Armes de Cac" (the same as the selector in the weapon), it doesn't work anymore.

**Comment by superseva** [comment]
I am not sure about that module "good translation module".
It's probably breaking the system in some way
What happens with no module ?

**Comment by superseva** [comment]
also what do you mean by
`update the default label from "Melee Weapons" to "Armes de Cac"` ?

**Comment by superseva** [comment]
Compendium is not translated.
All I did was changed Eng to FR in Foundry settings. And all of the FR translation is in the fr.json file that comes with the system. You can find it in the Foundry system folder
`systems/fallout2d20/lang/fr.json`

**Comment by extazz17** [comment]
So i set the language core to FRENCH.
Even if the labels are not the same, roll is ok
![image](https://github.com/superseva/fallout/assets/55485926/59967633-f084-4d04-a4da-104f77fbc2f4)
![image](https://github.com/superseva/fallout/assets/55485926/b01e445e-988c-43a2-a951-c5ee9f3c33b5)
![image](https://github.com/superseva/fallout/assets/55485926/4cac9755-b0f3-44fb-92a0-26442dff7487)

BUT, if i update the label of "Melee Weapons" and i set "Armes de CAC" (like the selector in the weapon card), the roll doesn't work anymore.
![image](https://github.com/superseva/fallout/assets/55485926/8b59ac6f-50c9-484e-bce8-b20ba636d1ea)
![image](https://github.com/superseva/fallout/assets/55485926/1e8266cd-2d8a-4cf0-847a-fa080002b3c4)

So i don't understand why it's ok on your side, i don't have any module launched.

**Comment by superseva** [comment]
Tou should not change compendium entries. I suggest reinstalling the system if you changed compendium. And then just switch foundry to FR

**Comment by extazz17** [comment]
Created a PR for this.

**Comment by superseva** [comment]
I'll check it out over the weekend.
I usually test it locally by creating a sym link to the folder by using (mklink command)
`https://www.howtogeek.com/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/`
or by simply copying my edited (local) version of `fallout` folder in to systems/fallout in foundry

What cloud you are using ?

**Comment by superseva** [comment]
hmmm Foundry doesn't have a cloud
you mean Forge ?

**Comment by extazz17** [comment]
Yes sorry it's not me who setup Foundry, I adapted your system to help a friend

**Comment by superseva** [comment]
I believe he need to upload edited system as his own custom system. I don't use it but there should be a button to upload your custom system.
Also, are kilos added in French book ? I see you multiply the weight by 5 for it. Is that in the fr version of book ?

**Comment by extazz17** [comment]
I checked on my side but there is no possibility to update files : https://forums.forge-vtt.com/t/ssh-ftp-and-the-forge/15308
Yes i have the FRENCH version of the book, here is the text in the PDF : 
![image](https://github.com/superseva/fallout/assets/55485926/270af89b-c647-406f-a769-310940917a42)

**Comment by extazz17** [comment]
I found a process to upload a custom system, but I'm afraid of breaking the current system and losing data.

**Comment by extazz17** [comment]
Ok I found how to do it, I use Google Chrome override files to manage this, everything is on the front side actually.
I successfully tested the management of the weight unit (with the settings) as well as the new way of rolling the weapon type.
So the PR will not be hard to validate.

**Comment by superseva** [comment]
Thanks, I'll do it over weekend when my head is in the right space :)

**Comment by superseva** [comment]
I am doing some testing and there is a problem with this solution when people use custom skill compendiums,(because there is no translation).
I am still not able to produce the initial problem you have with translation.
I'll need to find time to do more testing

**Comment by superseva** [comment]
I'll push the rest of the code for now (weight and such),but I'll keep the weapon/skill part of the code for now since none of the other FR players have reported that problem.

**Comment by Teylah57** [comment]
there is an old module for French translation on Foundry, don't use it, it break all the Fallout system.

**Comment by extazz17** [comment]
> I am doing some testing and there is a problem with this solution when people use custom skill compendiums,(because there is no translation). I am still not able to produce the initial problem you have with translation. I'll need to find time to do more testing

I don't understand because if a user uses a custom skill from the compendium, that means that it is not managed in your structure for the choice of weapon type so it doesn't impact the calcul for the roll.

All weapons types (in the weapon card) are hardcoded in `config.mjs` as `FALLOUT.WEAPONS.weaponType.[x]`.
The link will works with a link 1:1 on the translated strings like 
`"energyWeapons": "FALLOUT.WEAPONS.weaponType.energyWeapons",`

```
// item.system.weaponType = energyWeapons
skillName = game.i18n.localize(
    `FALLOUT.WEAPONS.weaponType.${item.system.weaponType}`,
)
```
In this exemple, the the i18n will search for a string in the current .json language with the same name, if : 
`the label of compendium skill name = label of weaponType`, the match will works.

So if an user create a new skill like "Super Mega Strength", this one will not be in the `FALLOUT.WEAPONS.weaponType` object so it doesn't impact the roll.
=> `let skillItem = item.actor.items.find((i) => i.name == skillName)` skillItem will be undefined

_Note : For my case, i don't have any good value in compendium, so i create a new skill and if I put exactly the same string as in the weapon type selector on the weapon sheet, the link works... That's how my PR works._

**Comment by extazz17** [comment]
For your information, i updated the module with the last version, i have now translated skills in new cards and the link works for the roll.
The compendium is still in English but everything works.
So either you can allow the PR for this, or you give up (I guess I was an isolated case).
However, you can accept new, modified translations.

**Comment by superseva** [comment]
We are in the process of transferring ownership of a system to another developer. In a few day the new repo will be activated and the issues will be transferred to them.
